onnxruntime/test/shared_lib/test_nontensor_types.cc

```diff
@@ -192,7 +194,7 @@ TEST(CApiTest, TypeInfoMap) {
   // ASSERT_EQ(map_value_type_info.GetTensorTypeAndShapeInfo().GetShape(), dims);
   ASSERT_EQ(map_value_type_info.GetTensorTypeAndShapeInfo().GetElementType(), ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT);
 
-  map_value_type_info.release();
+  //As it doesn't own the pointer, we should release the ownership, otherwise the pointer will be double freed.
```

Comment:
We can avoid this explicit release() call by just changing `Ort::MapTypeInfo` to `auto` on line 182  (seems like the Unowned class destructor calls Release() for us) ? Same for the sequence tensor example below. This keeps the tests consistent with GetTensorTypeAndShapeInfo() tests...

Reply:
I'm not sure, I'm not familiar with this piece of code. I would like to leave the refactoring work to the original author in case of I accidently broke something.

Reply:
I verified it and it keeps the code consistent. Would you like me to send out a PR for this ?